codegen/sr.rs

```diff
@@ -260,14 +254,11 @@ pub fn gen_sr_type_creation(grammar: &structs::Grammar) -> String {
                 quote! { {#( #init_list ),*} }
             };
             quote! {
-                pub fn #func_name #param_list -> TypeToken {
-                    let t = Type { ty: TypeEnum::#symbol #init_list, decorations: Vec::new() };
-                    if let Some(index) = self.types.iter().position(|x| *x == t) {
-                        TypeToken::new(index)
-                    } else {
-                        self.types.push(t);
-                        TypeToken::new(self.types.len() - 1)
-                    }
+                pub fn #func_name #param_list -> Token<Type> {
```

Comment:
Here too. It would be nice to have a separate CL for this change and the autogen touched files.